src/test/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensionRoutesSpec.scala

```diff
@@ -177,4 +181,138 @@ class GoogleExtensionRoutesSpec extends FlatSpec with Matchers with ScalatestRou
+    val samRoutes = new TestSamRoutes(mockResourceService, new UserService(directoryDAO, googleExt), new StatusService(directoryDAO, NoExtensions), UserInfo("", defaultUserInfo.userId, defaultUserInfo.userEmail, 0), directoryDAO) with GoogleExtensionRoutes {
+      val googleExtensions = googleExt
+    }
+
+    //create user
+    Post("/register/user") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Created
+    }
+
+    // create a pet service account
+    Get("/api/google/user/petServiceAccount/myproject") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      val response = responseAs[WorkbenchEmail]
+      response.value should endWith (s"@myproject.iam.gserviceaccount.com")
+    }
+
+    // create a pet service account key
+    Get("/api/google/user/petServiceAccount/myproject/key") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      val response = responseAs[ServiceAccountKeyWithEmail]
+      assert(response.validBefore.isDefined)
+    }
+  }
+
+  "DELETE /api/google/user/petServiceAccount/{project}/key/{keyId}" should "204 when deleting a key" in {
+    val resourceTypes = Map(resourceType.name -> resourceType)
+    val defaultUserInfo = UserInfo("accessToken", WorkbenchUserId("user1"), WorkbenchEmail("user1@example.com"), 0)
+    val googleDirectoryDAO = new MockGoogleDirectoryDAO()
+    val directoryDAO = new MockDirectoryDAO()
+    val googleIamDAO = new MockGoogleIamDAO()
+    val policyDAO = new MockAccessPolicyDAO()
+    val pubSubDAO = new MockGooglePubSubDAO()
+    val googleExt = new GoogleExtensions(directoryDAO, policyDAO, googleDirectoryDAO, pubSubDAO, googleIamDAO, googleServicesConfig, petServiceAccountConfig, configResourceTypes(CloudExtensions.resourceTypeName))
+
+    val mockResourceService = new ResourceService(resourceTypes, policyDAO, directoryDAO, googleExt, "example.com")
+    val samRoutes = new TestSamRoutes(mockResourceService, new UserService(directoryDAO, googleExt), new StatusService(directoryDAO, NoExtensions), UserInfo("", defaultUserInfo.userId, defaultUserInfo.userEmail, 0), directoryDAO) with GoogleExtensionRoutes {
+      val googleExtensions = googleExt
+    }
+
+    //create user
+    Post("/register/user") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Created
+    }
+
+    // create a pet service account
+    Get("/api/google/user/petServiceAccount/myproject") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      val response = responseAs[WorkbenchEmail]
+      response.value should endWith (s"@myproject.iam.gserviceaccount.com")
+    }
+
+    // create a pet service account key
+    Get("/api/google/user/petServiceAccount/myproject/key") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      val response = responseAs[ServiceAccountKeyWithEmail]
+      assert(response.validBefore.isDefined)
+    }
+
+    // create a pet service account key
```

Comment:
Delete, not create.

Also, everything above this point appears to be the same as the create test. It would be helpful to extract one or more functions here to make that more apparent... or more importantly make any differences (or lack thereof) apparent. `setupPetSATest` below is a great example of making the individual test cases more readable.